File: DirectoryListing/app/src/main/java/com/example/directorylisting/shared/AppManager.java
package com.example.directorylisting.shared;

import android.app.ProgressDialog;
import android.content.Context;

import com.bumptech.glide.load.Key;
import com.bumptech.glide.signature.StringSignature;
import com.example.directorylisting.api.WebService;
import com.example.directorylisting.entities.Individual;

/**
 * Created by Michael Steele on 3/22/17.
 * Copyright © 2017 Michael Steele. All rights reserved.
 */

public class AppManager extends Object {

    static public AppManager shared = new AppManager();

    public WebService webService = null;

    public String hostname = "localhost";
    public boolean allowInvalidCert = true;
    public String appUser = "test"; // TODO: Add login interface when auth is required (only if server prompts).
    public String appPassword = "test"; // TODO: Add login interface when auth is required (only if server prompts).

    public String baseURL = "https://" + hostname + ":8443/api/";
    public String invalidCertURL = "http://" + hostname + ":8080/static/server.crt";

    public Boolean directoryListingRefreshNeeded = false;

    public int imageCacheSize = 100 * 1024 * 1024;
    public int imageCompressionPercentage = 1;

    public ProgressDialog progressDialog = null;

    public static String packageName = null;

    public AppManager init(Context context, WebService.BasicSuccessFailureInterface basicSuccessFailureInterface) {
        webService = new WebService();
        webService.initService(context, 0, basicSuccessFailureInterface);

        return this;
    }

    /*
    Glide 4
    public Key getCacheKey(Individual individual) {
        return new ObjectKey(individual.imageCheck);
    }*/


    //Glide 3
    public Key getCacheKey(Individual individual) {
        return new StringSignature(individual.imageCheck);
    }

    public void showprogressDialog(Context context, String title, String message) {
        Boolean showDialog = false;

        if (progressDialog == null) {
            progressDialog = new ProgressDialog(context);
            showDialog = true;
        }

        progressDialog.setTitle(title);
        progressDialog.setMessage(message);
        progressDialog.setCancelable(false);

        if (showDialog == true) {

            progressDialog.show();
        }
    }

    public void dismissProgressDialog() {
        if (progressDialog == null) {
            return;
        }
        progressDialog.dismiss();
        progressDialog = null;
    }

}

File: DirectoryListing/app/src/main/java/com/example/directorylisting/shared/AppManager.kt
package com.example.directorylisting.shared

import android.app.ProgressDialog
import android.content.Context

import com.bumptech.glide.load.Key
import com.bumptech.glide.signature.StringSignature
import com.example.directorylisting.api.WebService
import com.example.directorylisting.entities.Individual

/**
 * Created by Michael Steele on 3/22/17.
 * Copyright © 2017 Michael Steele. All rights reserved.
 */

class AppManager : Any() {

    var webService: WebService = WebService()

    var hostname = "localhost"
    var allowInvalidCert = true
    var appUser = "test" // TODO: Add login interface when auth is required (only if server prompts).
    var appPassword = "test" // TODO: Add login interface when auth is required (only if server prompts).

    var baseURL = "https://$hostname:8443/api/"
    var invalidCertURL = "http://$hostname:8080/static/server.crt"

    var directoryListingRefreshNeeded: Boolean = false

    var imageCacheSize = 100 * 1024 * 1024
    var imageCompressionPercentage = 1

    var progressDialog: ProgressDialog? = null

    fun init(context: Context, basicSuccessFailureInterface: WebService.BasicSuccessFailureInterface): AppManager {
        webService.initService(context, 0, basicSuccessFailureInterface)

        return this
    }

    /*
    Glide 4
    public Key getCacheKey(Individual individual) {
        return new ObjectKey(individual.imageCheck);
    }*/


    //Glide 3
    fun getCacheKey(individual: Individual): Key {
        return StringSignature(individual.imageCheck)
    }

    fun showprogressDialog(context: Context, title: String, message: String) {
        var showDialog: Boolean = false

        showDialog = if (progressDialog == null) true else false

        val dialog = progressDialog?: ProgressDialog(context)

        dialog.setTitle(title)
        dialog.setMessage(message)
        dialog.setCancelable(false)

        if (showDialog == true) {
            dialog.show()
            progressDialog = dialog
        }
    }

    fun dismissProgressDialog() {
        val dialog = progressDialog?: return

        dialog.dismiss()

        progressDialog = null
    }

    companion object {

        var shared = AppManager()

        var packageName: String? = null
    }

}
